### PR TITLE
Added support to Microsoft.Extensions.Azure for configuring `managedIdentityObjectId`

### LINKED ISSUE #45661: [FEATURE REQ] Support object ID for ManagedIdentityCredential in AzureComponentFactory
### Library name

Microsoft.Extensions.Azure

### Please describe the feature.

The Azure Identity library's `ManagedIdentityCredential` is being enhanced to support object ID for user-assigned managed identity. See https://github.com/Azure/azure-sdk-for-net/pull/45605. Object ID is an alternative to using client ID or resource ID. With regards to `ManagedIdentityCredential`, we should have feature parity between Azure.Identity and Microsoft.Extensions.Azure. It would involve updating this code: https://github.com/Azure/azure-sdk-for-net/blob/91dc3f14f2d219c61ec7cdb715764becfce8fc73/sdk/extensions/Microsoft.Extensions.Azure/src/Internal/ClientFactory.cs#L115-L128

From a customer's perspective, they should be able to provide configuration like the following:

```json
{
    "credential": "managedidentity",
    "managedIdentityObjectId":  "<managedIdentityObjectId>"
}
```

Since we don't support object ID via `DefaultAzureCredential`, the following code flow is unaffected: https://github.com/Azure/azure-sdk-for-net/blob/91dc3f14f2d219c61ec7cdb715764becfce8fc73/sdk/extensions/Microsoft.Extensions.Azure/src/Internal/ClientFactory.cs#L218-L249

### LINKED ISSUE #45806: [BUG] net-core-ci failure
### Library name and version

Microsoft.Extensions.Azure

### Describe the bug

In https://github.com/Azure/azure-sdk-for-net/pull/45617, net-core-ci is failing 

> ##[error]sdk/extensions/Microsoft.Extensions.Azure/tests/ClientFactoryTests.cs(340,65): Error CS1061: 'ManagedIdentityClient' does not contain a definition for 'ClientId' and no accessible extension method 'ClientId' accepting a first argument of type 'ManagedIdentityClient' could be found (are you missing a using directive or an assembly reference?)

### Expected behavior

net-core-ci should apss

### Actual behavior

net-core-ci is failing

### Reproduction Steps

https://dev.azure.com/azure-sdk/public/_build/results?buildId=4109288&view=logs&j=8f1d288f-f763-5c12-3b1b-d79c9adacead&t=1a409d55-c12a-5808-4a50-b8ba34872ec8&l=110

### Environment

_No response_

### PR DESCRIPTION
**Pull request by christothes**
closes #45806
closes #45661

### CODE REVIEW

File: sdk/extensions/Microsoft.Extensions.Azure/tests/ClientFactoryTests.cs [hunk 1]
```diff
@@ -299,10 +300,15 @@ public void CreatesDefaultAzureCredential(
...
+            if (objectId)
+            {
+                configEntries.Add(new KeyValuePair<string, string>("managedIdentityObjectId", "objectId"));
+            }
+
             IConfiguration configuration = new ConfigurationBuilder().AddInMemoryCollection(configEntries).Build();
 
             // if both clientId and resourceId set, we expect an ArgumentException
-            if (clientId && resourceId)
+            if ((clientId && resourceId) || objectId)
```
Comment (annelo-msft) [inline sdk/extensions/Microsoft.Extensions.Azure/tests/ClientFactoryTests.cs:311]: Would it be possible to update the comment, or move the check to a separate block?  Compound booleans are hard to mentally parse for humans to understand the code and know exactly what it's testing to make sure it's correct, or use a test failure to figure out what regressed in the code being validated by the test.

File: sdk/extensions/Microsoft.Extensions.Azure/tests/ClientFactoryTests.cs [hunk 2]
```diff
@@ -338,18 +344,27 @@ public void CreatesDefaultAzureCredential(
 
             // TODO: Since these can't build with project reference, we have to comment them out for now.
             // When we resolve https://github.com/Azure/azure-sdk-for-net/issues/45806, we can add them back.
```
Comment (annelo-msft) [inline sdk/extensions/Microsoft.Extensions.Azure/tests/ClientFactoryTests.cs:346]: If we're ready to uncomment this, should we change the comment?

File: sdk/extensions/Microsoft.Extensions.Azure/tests/ClientFactoryTests.cs [hunk 3]
```diff
@@ -362,14 +377,15 @@ public void CreatesManagedServiceIdentityCredentialsWithClientId()
             Assert.IsInstanceOf<ManagedIdentityCredential>(credential);
             var managedIdentityCredential = (ManagedIdentityCredential)credential;
 
-            var client = (ManagedIdentityClient)typeof(ManagedIdentityCredential).GetProperty("Client", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(managedIdentityCredential);
-            var clientId = typeof(ManagedIdentityClient).GetProperty("ClientId", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(client);
+            string clientId;
+            int idType;
+            ReflectIdAndType(managedIdentityCredential, out clientId, out idType);
```
Comment (annelo-msft) [inline sdk/extensions/Microsoft.Extensions.Azure/tests/ClientFactoryTests.cs:382]: Curiosity only: why do we need reflection here?  Can we not make a runtime type check?

File: sdk/extensions/Microsoft.Extensions.Azure/tests/ClientFactoryTests.cs (hunk 2 above)
Comment (christothes) [reply sdk/extensions/Microsoft.Extensions.Azure/tests/ClientFactoryTests.cs:346]: thanks - forgot to delete the comment!

File: sdk/extensions/Microsoft.Extensions.Azure/tests/ClientFactoryTests.cs (hunk 1 above)
Comment (christothes) [reply sdk/extensions/Microsoft.Extensions.Azure/tests/ClientFactoryTests.cs:311]: updated the comment.

File: sdk/extensions/Microsoft.Extensions.Azure/tests/ClientFactoryTests.cs (hunk 3 above)
Comment (christothes) [reply sdk/extensions/Microsoft.Extensions.Azure/tests/ClientFactoryTests.cs:382]: Because these properties and fields are internal or private.

**Review by jsquire — APPROVED** [review]

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/extensions/Microsoft.Extensions.Azure/CHANGELOG.md [hunk 4]
```diff
@@ -4,7 +4,7 @@
 
 ### Features Added
 
-- Added support for constructing a `DefaultAzureCredential` or `ManagedIdentityCredential` from config by setting the `managedIdentityObjectId` key.
+- Added support for constructing a `ManagedIdentityCredential` from config by setting the `managedIdentityObjectId` key.
```
Comment (annelo-msft) [inline sdk/extensions/Microsoft.Extensions.Azure/CHANGELOG.md:7]: Mostly just curious here -- it looks like you changed the changelog -- was it incorrect before?  Did we add support for both or remove one?

**Review by annelo-msft — APPROVED** [review]

File: sdk/extensions/Microsoft.Extensions.Azure/CHANGELOG.md (hunk 4 above)
Comment (christothes) [reply sdk/extensions/Microsoft.Extensions.Azure/CHANGELOG.md:7]: We added support for objectId to ManagedIdentityCredential, but not DAC. Your comment about the test logic reminded me that what I originally said here was incorect for DAC.

**Comment by christothes** [comment]
@microsoft-github-policy-service rerun

**Pull request merged**